### DISCUSSION TITLE
ANCOM

### DISCUSSION THREAD

**Discussion by johdoh** [scikit-bio/scikit-bio#2044] [Q&A]
When trying to calculate ancom using a otu table with 95 rows (samples) × 4160 columns (OTUs) and 4 groups. i get the warning: "ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite es = hypotest_fun_out(*samples, **kwds)".  Calculation also seems to be stuck and delivers no results within 2h on a 12th Gen Intel Core i7-12700H with 64Gb of Ram. 

I had added a pseudocount of 0.001 as suggested and when checking for duplicated rows in the otu table (after droping sample labels) no duplicated rows are found. Any ideas?

**Comment by qiyunzhu** [comment]
@johdoh Thanks for the question. @mortonjt Do you have some thoughts?

**Comment by mortonjt** [comment]
Regarding runtime, I recommend subsetting the features to debug. ANCOM
scales quadratically, so if you have >10K features it'll take >8 hours

Regarding the actual error, the most likely culprit is zeros. If you have
features that are all zeros (or have all the same values in general),
you’ll get this error
